versioned_plugins.rb

```diff
@@ -320,16 +320,7 @@ def extract_doc(doc, plugin_full_name, release_tag, release_date, changelog_url)
         .gsub(":include_path: ../../../logstash/docs/include", ":include_path: ../include/6.x")
         .gsub(/[\t\r ]+$/,"")
 
-      content = content
-        .gsub("<<string,string>>", "{logstash-ref}/configuration-file-structure.html#string[string]")
-        .gsub("<<array,array>>", "{logstash-ref}/configuration-file-structure.html#array[array]")
-        .gsub("<<number,number>>", "{logstash-ref}/configuration-file-structure.html#number[number]")
-        .gsub("<<boolean,boolean>>", "{logstash-ref}/configuration-file-structure.html#boolean[boolean]")
-        .gsub("<<hash,hash>>", "{logstash-ref}/configuration-file-structure.html#hash[hash]")
-        .gsub("<<password,password>>", "{logstash-ref}/configuration-file-structure.html#password[password]")
-        .gsub("<<path,path>>", "{logstash-ref}/configuration-file-structure.html#path[path]")
-        .gsub("<<uri,uri>>", "{logstash-ref}/configuration-file-structure.html#uri[uri]")
-        .gsub("<<bytes,bytes>>", "{logstash-ref}/configuration-file-structure.html#bytes[bytes]")
+      content = fix_configuration_file_structure_links(content)
```

Comment:
Would you find this more readable if you had an "intercept_anchors" helper that took the content, a list of anchors to intercept, and the page to re-route them to like:

```ruby
      content = intercept_anchors(content, %w(string array number boolean hash password path uri bytes), "{logstash-ref}/configuration-file-structure.html")
```

Reply:
Not sure we want to make deviating from the standard so easy.  If we add a new input type, we should vet it and possibly update the lists of inputs. 

https://github.com/logstash-plugins/logstash-filter-elasticsearch/pull/166#discussion_r1132817389